app/models/server/models/Subscriptions.js

```diff
@@ -1429,6 +1429,25 @@ export class Subscriptions extends Base {
 
 		return this.update(query, update);
 	}
+
+	deleteAllThreadsByRoomId(rid, ts, users) {
+		const query = {
+			rid,
+			ts,
+		};
+
+		if (users.length) {
+			query['u.username'] = { $in: users };
+		}
+
+		const update = {
+			$set: {
+				tunread: [],
```

Comment:
you shouldn't cleanup *all* unread threads, as some threads could not have been deleted by the purge criterias. so instead you should remove from the array only the threads that got removed by the [remove command](https://github.com/RocketChat/Rocket.Chat/pull/18944/files#diff-1224c98fd028f6bac9161077e73f81c3R43)

Reply:
hum, my bad. It seems I got confused about how mongo queries work, but I will fix it.

Reply:
np.. you'll probably need to use `$pullAll`: https://docs.mongodb.com/manual/reference/operator/update/pullAll/